cmd/nerdctl/volume_ls.go

```diff
@@ -54,220 +45,63 @@ func newVolumeLsCommand() *cobra.Command {
+	options.Quiet = quiet
+	format, err := cmd.Flags().GetString("format")
 	if err != nil {
 		return err
 	}
-	if quiet && volumeSize {
-		logrus.Warn("cannot use --size and --quiet together, ignoring --size")
-		volumeSize = false
+	options.Format = format
+	size, err := cmd.Flags().GetBool("size")
+	if err != nil {
+		return err
 	}
+	options.Size = size
 	filters, err := cmd.Flags().GetStringSlice("filter")
 	if err != nil {
 		return err
 	}
-	labelFilterFuncs, nameFilterFuncs, sizeFilterFuncs, isFilter, err := getVolumeFilterFuncs(filters)
+	options.Filters = filters
+	ns, err := cmd.Flags().GetString("namespace")
 	if err != nil {
 		return err
 	}
-	if len(sizeFilterFuncs) > 0 && quiet {
-		logrus.Warn("cannot use --filter=size and --quiet together, ignoring --filter=size")
-		sizeFilterFuncs = nil
-	}
-	if len(sizeFilterFuncs) > 0 && !volumeSize {
-		logrus.Warn("should use --filter=size and --size together")
-		cmd.Flags().Set("size", "true")
-		volumeSize = true
-	}
-	w := cmd.OutOrStdout()
-	var tmpl *template.Template
-	format, err := cmd.Flags().GetString("format")
+	options.Namespace = ns
+	dataRoot, err := cmd.Flags().GetString("data-root")
 	if err != nil {
 		return err
 	}
-	switch format {
-	case "", "table", "wide":
-		w = tabwriter.NewWriter(cmd.OutOrStdout(), 4, 8, 4, ' ', 0)
-		if !quiet {
-			if volumeSize {
-				fmt.Fprintln(w, "VOLUME NAME\tDIRECTORY\tSIZE")
-			} else {
-				fmt.Fprintln(w, "VOLUME NAME\tDIRECTORY")
-			}
-		}
-	case "raw":
-		return errors.New("unsupported format: \"raw\"")
-	default:
-		if quiet {
-			return errors.New("format and quiet must not be specified together")
-		}
-		var err error
-		tmpl, err = formatter.ParseTemplate(format)
-		if err != nil {
-			return err
-		}
-	}
-
-	vols, err := getVolumes(cmd)
+	options.DataRoot = dataRoot
+	address, err := cmd.Flags().GetString("address")
 	if err != nil {
 		return err
 	}
-
-	for _, v := range vols {
-		if isFilter && !volumeMatchesFilter(v, labelFilterFuncs, nameFilterFuncs, sizeFilterFuncs) {
-			continue
-		}
-		p := volumePrintable{
-			Driver:     "local",
-			Labels:     "",
-			Mountpoint: v.Mountpoint,
-			Name:       v.Name,
-			Scope:      "local",
-		}
-		if v.Labels != nil {
-			p.Labels = formatter.FormatLabels(*v.Labels)
-		}
-		if volumeSize {
-			p.Size = progress.Bytes(v.Size).String()
-		}
-		if tmpl != nil {
-			var b bytes.Buffer
-			if err := tmpl.Execute(&b, p); err != nil {
-				return err
-			}
-			if _, err = fmt.Fprintf(w, b.String()+"\n"); err != nil {
-				return err
-			}
-		} else if quiet {
-			fmt.Fprintln(w, p.Name)
-		} else if volumeSize {
-			fmt.Fprintf(w, "%s\t%s\t%s\n", p.Name, p.Mountpoint, p.Size)
-		} else {
-			fmt.Fprintf(w, "%s\t%s\n", p.Name, p.Mountpoint)
-		}
-	}
-	if f, ok := w.(formatter.Flusher); ok {
-		return f.Flush()
-	}
-	return nil
+	options.Address = address
+	return volume.Ls(options)
 }
 
 func getVolumes(cmd *cobra.Command) (map[string]native.Volume, error) {
```

Comment:
is this `getVolumes` func still being used somewhere? This func seems no longer necessary since it only read 4 flags from cmd and call another func.

If it's not common enough (e.g., used a lot by other commands as well), then we should remove this helper and instead prefer calling `volume.Volumes` directly.